service/resource/project/resource.go

```diff
@@ -73,105 +90,140 @@ func (r *Resource) GetDesiredState(obj interface{}) (interface{}, error) {
+	}
 
-	return nil, nil
+	return desiredProjects, nil
 }
 
 func (r *Resource) GetCreateState(obj, currentState, desiredState interface{}) (interface{}, error) {
-	customObject, err := toCustomObject(obj)
+	currentProjects, err := toProjects(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredProjects, err := toProjects(desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	r.logger.Log("debug", "get create state")
-
-	r.logger.Log("TODO", fmt.Sprintf("implement logic based on received custom object: %#v", customObject))
+	var projectsToCreate []Project
 
-	r.logger.Log("debug", fmt.Sprintf("found create state: %#v", nil))
+	for _, desiredProject := range desiredProjects {
+		if !existsProjectByName(currentProjects, desiredProject.Name) {
+			projectsToCreate = append(projectsToCreate, desiredProject)
+		}
+	}
 
-	return nil, nil
+	return projectsToCreate, nil
 }
 
 func (r *Resource) GetDeleteState(obj, currentState, desiredState interface{}) (interface{}, error) {
-	customObject, err := toCustomObject(obj)
+	currentProjects, err := toProjects(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredProjects, err := toProjects(desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	r.logger.Log("debug", "get delete state")
-
-	r.logger.Log("TODO", fmt.Sprintf("implement logic based on received custom object: %#v", customObject))
+	var projectsToDelete []Project
 
-	r.logger.Log("debug", fmt.Sprintf("found delete state: %#v", nil))
+	for _, currentProject := range currentProjects {
+		if existsProjectByName(desiredProjects, currentProject.Name) {
+			projectsToDelete = append(projectsToDelete, currentProject)
+		}
+	}
 
-	return nil, nil
+	return projectsToDelete, nil
 }
 
 func (r *Resource) GetUpdateState(obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
-	customObject, err := toCustomObject(obj)
+	currentProjects, err := toProjects(currentState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	desiredProjects, err := toProjects(desiredState)
 	if err != nil {
 		return nil, nil, nil, microerror.Mask(err)
 	}
 
-	r.logger.Log("debug", "get delete state")
+	var projectsToUpdate []Project
 
-	r.logger.Log("TODO", fmt.Sprintf("implement logic based on received custom object: %#v", customObject))
+	for _, desiredProject := range desiredProjects {
+		if !existsProjectByName(currentProjects, desiredProject.Name) {
+			continue
+		}
 
-	r.logger.Log("debug", fmt.Sprintf("found delete state: %#v", nil))
+		currentProject, err := getProjectByName(currentProjects, desiredProject.Name)
+		if err != nil {
+			return nil, nil, nil, microerror.Mask(err)
+		}
+
+		if currentProject.Ref != desiredProject.Ref {
+			projectsToUpdate = append(projectsToUpdate, desiredProject)
+		}
+	}
 
-	return nil, nil, nil, nil
+	return []Project{}, []Project{}, projectsToUpdate, nil
```

Comment:
The update procedure of the project resource cannot create or delete projects. Thus we only handle the updates here. The installer implementation is a bit "dumb" and only provides one method that installs and updates in one command. This could be separated later but I am not too fuzzed about this now.